patches/2024.1/neutron/backport-928288.patch

```diff
@@ -0,0 +1,709 @@
+From 17541a43aa56878107a54b4658fb256c4432b38d Mon Sep 17 00:00:00 2001
+From: Ihar Hrachyshka <ihrachys@redhat.com>
+Date: Fri, 16 Aug 2024 22:22:24 +0000
+Subject: [PATCH] Support nested SNAT for ml2/ovn
+
+When ovn_router_indirect_snat = True, ml2/ovn will set a catch-all snat
+rule for each external ip, instead of a snat rule per attached subnet.
+
+NB: This option is global to cluster and cannot be controlled per
+project or per router.
+
+NB2: this patch assumes that 0.0.0.0/0 snat rules are properly handled
+by OVN. Some (e.g. 22.03 and 24.03) OVN versions may have this scenario
+broken. See: https://issues.redhat.com/browse/FDP-744 for details.
```

Comment:
I can't seem to find out which versions of OVN would not be broken. Are you sure that this would not affect us?

Reply:
The default is `ovn_router_indirect_snat = False`  which should keep the current behaviour.

Reply:
Looks like the crucial patch is included in OVN versions v24.09.1 and v24.09.0: https://github.com/ovn-org/ovn/commit/40136a2f

Reply:
We have OVN 24.03 in OSISM 8.0.2. The OVN version will not be changed in the next months.

Reply:
Would you rather wait with adding this backport until you can switch to v24.09.x or is it maybe possible to add the backport and let us use the newer version already via `images.yml`? Or are the images not even existent?

Reply:
so to me this sounds like adding this backport is useless, then?

Reply:
> Would you rather wait with adding this backport until you can switch to v24.09.x or is it maybe possible to add the backport and let us use the newer version already via `images.yml`? Or are the images not even existent?

OVN 24.03 is a LTS release of OVN. 24.09 is the latest OVN release. I think it's a bad idea to use a OVN release that's newer than the OpenStack Neutron release you have in use.

Reply:
Okay then we could/should probably wait until v24.09.x is usable.